ci/pipeline.yml

```diff
@@ -0,0 +1,305 @@
+    #       <<: *staging-cf
+    #       CF_APP_NAME: docker-registry-mirror
+  on_failure:
+    put: slack
+    params: &slack-params
+      text: |
+        :x: FAILED to deploy docker registry mirror to staging. Deployment was not canceled.
+        <$ATC_EXTERNAL_URL/teams/$BUILD_TEAM_NAME/pipelines/$BUILD_PIPELINE_NAME/jobs/$BUILD_JOB_NAME/builds/$BUILD_NAME|View build details>
+      channel: ((slack-channel))
+      username: ((slack-username))
+      icon_url: ((slack-icon-url))  
+
+- name: deploy-proxy-staging
+  serial_groups: [proxy]
+  plan:
+  - in_parallel:
+    - get: nginx-conf
+      trigger: true
+    - get: proxy-source
+      trigger: true
+      params: {depth: 1}
+    - get: ci-source
+      params: {depth: 1}
+  - task: deploy-proxy
+    file: ci-source/ci/tasks/deploy-proxy.yml
+    tags: [iaas]
+    params: 
+      <<: *staging-cf
+    # Don't cancel the deployment in staging so we can debug the failure
+    # on_failure:
+    #   try: 
+    #     task: cancel-deployment
+    #     file: ci-source/ci/tasks/cancel-deployment.yml
+    #     params:
+    #       <<: *staging-cf
+    #       CF_APP_NAME: docker-registry-proxy
+  on_failure:
+    put: slack
+    params:
+      <<: *slack-params
+      text: |
+        :x: FAILED to deploy docker registry mirror proxy to staging. Deployment was not canceled.
+        <$ATC_EXTERNAL_URL/teams/$BUILD_TEAM_NAME/pipelines/$BUILD_PIPELINE_NAME/jobs/$BUILD_JOB_NAME/builds/$BUILD_NAME|View build details>
+
+- name: integration-test
+  serial_groups: [mirror,proxy]
+  plan:
+  - in_parallel:
+    - get: registry-source
+      trigger: true
+      params: {depth: 1}
+      passed: [deploy-registry-staging]
+    - get: docker-registry-image
+      trigger: true
+      passed: [deploy-registry-staging]
+    - get: nginx-conf
+      trigger: true
+      passed: [deploy-proxy-staging]
+    - get: proxy-source
+      trigger: true
+      params: {depth: 1}
+      passed: [deploy-proxy-staging]
+    - get: ci-source
+      params: {depth: 1}
+  # Be sure we can pull an image through the mirror using the docker-image resource type
+  - task: test-pull-succeeds-docker-image
+    config:
+      platform: linux
+      image_resource:
+        type: docker-image
+        source:
+          repository: 18fgsa/concourse-task
+          registry_mirror: "https://((staging-mirror-hostname)).app.cloud.gov:443"
+      run:
+        path: /bin/sh
+        args:
+        - -ec
+        - |
+          date
+  # Be sure we can pull an image through the mirror using the registry-image resource type
+  - task: test-pull-succeeds-registry-image
+    config:
+      platform: linux
+      image_resource:
+        type: registry-image
+        source:
+          repository: 18fgsa/concourse-task
+          registry_mirror: 
+            host: "((staging-mirror-hostname)).app.cloud.gov:443"
+      run:
+        path: /bin/sh
+        args:
+        - -ec
+        - |
+          date
+    # on_failure:
```

Comment:
Should something happen here, or will that come later after the initial work?

Reply:
Extraneous, thanks. I removed this as there is an on_failure block at the job level.